src/lib/createStreamSink.js

```diff
@@ -25,10 +25,10 @@ function promiseToCallback(promise) {
   return function (cb) {
     promise.then(
       function (data) {
-        setImmediate(cb, null, data);
+        setTimeout(cb, 0, null, data);
```

Comment:
`setTimeout` is not the same as `setImmedate`. Actually I think none of them are needed here anyway. `Promise.then(...)` already creates a microtask. The same in `dnode-util`